frontend/src/App.tsx

```diff
@@ -28,12 +29,21 @@ const App = () => {
     dispatch(fetchStatus());
   }, [dispatch]);
 
+  const usersVisitationStatus = localStorage.getItem('users-status');
+
+  useEffect(() => {
+    if (usersVisitationStatus === null) {
+      setIsFirstVisit(true);
+    }
+    localStorage.setItem('users-status', JSON.stringify('user-has-visited'));
+  }, [setIsFirstVisit, usersVisitationStatus]);
+
```

Comment:
Do we need to rely on state here?   Can we just check what's in local storage in the places where we need to know if it's a first visit or not?

Reply:
Not sure at the moment.  It would be great to chat about this at some point today.